flink-metrics/flink-metrics-influxdb/src/test/java/org/apache/flink/metrics/influxdb/MeasurementInfoProviderTest.java

```diff
@@ -0,0 +1,133 @@
+	private static final Random RANDOM = new Random();
+
+	private final MeasurementInfoProvider provider = new MeasurementInfoProvider();
+
+	@Test
+	public void testGetMetricInfo() {
+		// MetricRegistry, required as the first argument in metric groups.
+		MetricRegistryImpl metricRegistry = new MetricRegistryImpl(
+			MetricRegistryConfiguration.fromConfiguration(new Configuration()));
+
+		// Create an example, nested metric group: taskmanager -> job -> task
+		// Variables: <host>, <tm_id>
+		String hostname = "loc<>al\"::host\".:";
+		String taskManagerId = "tas:kMana::ger";
+		TaskManagerMetricGroup tmMetricGroup = new TaskManagerMetricGroup(
+			metricRegistry, hostname, taskManagerId);
+
+		// Variables: <job_id>, <job_name>
+		JobID jobID = new JobID();
+		String jobName = "testJ\"ob:-!ax..?";
+		TaskManagerJobMetricGroup tmJobMetricGroup = new TaskManagerJobMetricGroup(
+			metricRegistry, tmMetricGroup, jobID, jobName);
+
+		// Variables: <task_id>, <task_attempt_id>, <task_name>, <subtask_index>, <task_attempt_num>
+		JobVertexID taskId = new JobVertexID();
+		AbstractID taskAttemptID = new AbstractID();
+		String taskName = "test\"Ta\"..sk";
+		int subtaskIndex = RANDOM.nextInt();
+		int taskAttemptNum = RANDOM.nextInt();
+		TaskMetricGroup taskMetricGroup = new TaskMetricGroup(
+			metricRegistry, tmJobMetricGroup, taskId, taskAttemptID, taskName, subtaskIndex, taskAttemptNum);
+
+		String metricName = "testCounter";
+		MetricGroup metricGroup = new FrontMetricGroup<>(0, taskMetricGroup);
+
+		MeasurementInfo info = provider.getMetricInfo(metricName, metricGroup);
+		assertNotNull(info);
+		assertEquals(
+			String.join("" + MeasurementInfoProvider.SCOPE_SEPARATOR, "taskmanager", "job", "task", metricName),
+			info.getName());
+		assertThat(info.getTags(), hasEntry("host", hostname));
+		assertThat(info.getTags(), hasEntry("tm_id", taskManagerId));
+		assertThat(info.getTags(), hasEntry("job_id", jobID.toString()));
+		assertThat(info.getTags(), hasEntry("job_name", jobName));
+		assertThat(info.getTags(), hasEntry("task_id", taskId.toString()));
+		assertThat(info.getTags(), hasEntry("task_attempt_id", taskAttemptID.toString()));
+		assertThat(info.getTags(), hasEntry("task_name", taskName));
+		assertThat(info.getTags(), hasEntry("subtask_index", String.valueOf(subtaskIndex)));
+		assertThat(info.getTags(), hasEntry("task_attempt_num", String.valueOf(taskAttemptNum)));
+		assertEquals(9, info.getTags().size());
+
+		try {
+			metricRegistry.shutdown().get();
+		} catch (InterruptedException | ExecutionException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Test
+	public void simpleTestGetMetricInfo() {
+		String logicalScope = "myService.Status.JVM.ClassLoader";
+		Map<String, String> variables = new HashMap<>();
+		variables.put("<A>", "a");
+		variables.put("<B>", "b");
+		variables.put("<C>", "c");
+		String metricName = "ClassesLoaded";
+		FrontMetricGroup metricGroup = mock(FrontMetricGroup.class);
```

Comment:
create a normal FrontMetricGroup that wraps an `UnregisteredOperatorMetricGroup` (or similar) with overridden methods `getAllVariables` and `getLocalScope`.

Reply:
Any arguments against using mock here?